From d615dc63bf104f93813e4c5eecea84ea0f7f8d2d Mon Sep 17 00:00:00 2001
From: Konstantin Porotchkin <kostap@marvell.com>
Date: Wed, 25 Jul 2018 17:19:28 +0300
Subject: [PATCH 0816/1200] fix: mmc: xenon: Correct the host speed
 configuration

Remove double assignment to maximum clock frequency of
SDHCI host and fix the zero clock value in call to
sdhci_setup_cfg() function.

Change-Id: I88df98e26e625abf38ce7e3893b52862e0444dc2
Signed-off-by: Konstantin Porotchkin <kostap@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/58410
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Igal Liberman <igall@marvell.com>
---
 drivers/mmc/xenon_sdhci.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/mmc/xenon_sdhci.c b/drivers/mmc/xenon_sdhci.c
index 6a8e428483..1fa1655dc5 100644
--- a/drivers/mmc/xenon_sdhci.c
+++ b/drivers/mmc/xenon_sdhci.c
@@ -500,8 +500,7 @@ static int xenon_sdhci_probe(struct udevice *dev)
 	host->ops = &xenon_sdhci_ops;
 	host->max_clk = XENON_MMC_MAX_CLK;
 
-	host->max_clk = XENON_MMC_MAX_CLK;
-	ret = sdhci_setup_cfg(&plat->cfg, host, 0, 0);
+	ret = sdhci_setup_cfg(&plat->cfg, host, XENON_MMC_MAX_CLK, 0);
 	if (ret)
 		return ret;
 
-- 
2.22.0

